README.md

```diff
@@ -3,8 +3,76 @@
+```
+plugins {
+    id("com.liftric.octopus-deploy-plugin") version "whatever"
+}
+[...]
+octopus {
+    serverUrl = "http://localhost:8080/"
+    apiKey = "API-TESTTEST123TRESDTSDD"
+
+    generateChangelogSinceLastTag = true
+
+    val jar by tasks.existing(Jar::class)
+    packageName = jar.get().archiveBaseName.get()
+    version = jar.get().archiveVersion.get()
+    pushPackage = jar.get().archiveFile.get().asFile
+}
+```
+
+See the example directory for a minimal, working example.
+
+## tasks
+The following tasks are added by the plugin:
+
+Task | Description
+---|---
+previousTag | Calls git describe to receive the previous tag name. Will fail if no tag is found.
+firstCommitHash | Calls git log to get the first commit hash of the current history tree
+commitsSinceLastTag | Calls git log to receive all commits since the previous tag or the first commit of the current history.
+createBuildInformation | Creates the octopus build-information file.
+uploadBuildInformation | Uploads the created octopus build-information file.
+uploadPackage | Uploads the package to octopus.
+
+For normale use-cases, only `uploadBuildInformation` are `uploadPackage` are needed to call explicitly. Depending
+task will be called implicitly by both as needed.
+
+**Noteworthy**: The build-information can be uploaded before the package itself. 
+Useful when creating automatic releases and using the commits in the release notes in octopus.
+
+## naming
+Octopus deploy expects the name and version in the following format: `<name>.<version>.<extension>`
+
+For Java archives, `<name>-<version>.<extension>` is conventional, so should be changed to get picked up properly by octopus.
+Otherwise the first version number part will be parsed as part of the name.
 
 ## configuration
+After the plugin is applied, the vault extension is registered with the following settings:
+
+Property | Description | default value 
+---|---|---
+apiKey | Octopus deploy server API key | -
+serverUrl | Octopus deploy server URL | -
+generateChangelogSinceLastTag | Enable to calculate the commits for the changelog when uploading build-information (needs git installed) | false
+commitLinkBaseUrl | Prefix / Baseurl for the build-information commit urls | http://git.example.com/repo/commits/
+outputDir | Output folder for files generated by the plugin | build/octopus
+gitRoot | Directory to run the git helpers in. By default the projects root dir | project.rootDir
+pushPackage | Target file (package) which will be uploaded to octopus | 
+version | Package version | -
+packageName | Package name | -
+buildInformationOverwriteMode | octo build-information OverwriteMode | -
+pushOverwriteMode | octo push OverwriteMode | -
+buildInformationAddition | Customize the final octopus build-information before uploading | {}
+gitlab() | Default `buildInformationAddition` implementation adding context from the CI environment for Gitlab CI. Also sets `commitLinkBaseUrl`. | not applied
```

Comment:
Just a small remark while looking at the code doesn't it make sense to also use a computed property for gitlab?

Reply:
What do you mean by that? 
`fun gitlab(): BuildInformation.() -> Unit`?

Feel free calling or sending via voice message 👯

Reply:
You used for every other property a computed property in the code. So I just thought why not using one for gitlab instead of using a function :) But wasn't really that important!